### PR TITLE
Add method to read and skim large TTrees

### PR DESCRIPTION
**Pull request by fmazzasc**

### CODE REVIEW

File: hipe4ml/analysis_utils.py [hunk 1]
```diff
@@ -181,7 +185,84 @@ def train_test_generator(data_list, labels_list, sliced_df=False, **kwds):
...
+        String containing the cuts to be applied as preselection on the data contained in the original
+        tree. The string syntax is the one required in the pandas.DataFrame.query() method.
+        You can refer to variables in the environment by prefixing them with an ‘@’ character like @a + b
+
+    model_handler: hipe4ml ModelHandler
+        Model handler to be applied as a preselection on the data contained in the original
+        tree. A column named model_output is added to the tree_handler
+
+    score_cut: int or float
+        Score to be applied as a starting preselection on the data contained in the original tree
```
Comment (fcatalan92) [inline hipe4ml/analysis_utils.py:220]: This does not support multiclass classifications. Maybe the code could be refactored adding another method for application on large files

**Review by fcatalan92 — CHANGES_REQUESTED** [review]

**Comment by fcatalan92** [comment]
Hi @fmazzasc, nice idea! In general, although, it feels a bit a hacky solution. Maybe we can improve the design a bit

**Comment by fmazzasc** [comment]
Hi @fcatalan92, I have solved your comments and added the support to multi-classification

**Comment by fcatalan92** [comment]
Thanks @fmazzasc! I would say that now it's ok. The only thing I do not like is that now the `tree_handler` is using a function from `analysis_utils` which is a module containing misc functions. `apply_model_handler_to_pandas` probably could be moved to the model_handler class.

**Comment by fmazzasc** [comment]
Thanks @fcatalan92 , you were right. I dropped the `apply_model_handler_to_pandas` method and added few lines directly in `get_handler_from_large_file`

File: hipe4ml/tree_handler.py [hunk 2]
```diff
@@ -73,6 +77,93 @@ def __len__(self):
...
+
+        result = []
+        for data in iterator:
+            if model_handler is not None:
+                predictions = model_handler.predict(data, output_margin)
+                n_classes = model_handler.get_n_classes()
+                if n_classes > 2:
+                    for i_class in range(n_classes):
+                        column_name = f'model_output_{i_class}'
+                        data[column_name[i_class]] = predictions[:, i_class]
```
Comment (fcatalan92) [inline hipe4ml/tree_handler.py:145]: Here `column_name` is a string, not a list.

File: hipe4ml/analysis_utils.py [hunk 3]
```diff
@@ -164,7 +163,8 @@ def train_test_generator(data_list, labels_list, sliced_df=False, **kwds):
         del data_list, data_frame
         df_tot_train_test = pd.concat(df_list, sort=True)
         del df_list
-        train_test = train_test_split(df_tot_train_test, np.array(labels_train_test), **kwds)
+        train_test = train_test_split(
+            df_tot_train_test, np.array(labels_train_test), **kwds)
```
Comment (fcatalan92) [inline hipe4ml/analysis_utils.py:167]: It looks like your pylint is not picking up the configuration file of the repo where the max line length is 120.

**Review by fcatalan92 — CHANGES_REQUESTED** [review]
I would remove all spurious change due to the wrong line length limit. Then, what about a test of this new functionality?

File: hipe4ml/analysis_utils.py (hunk 3 above)
Comment (fmazzasc) [reply hipe4ml/analysis_utils.py:167]: It was actually due to the vscode auto-formatter. Fixed now

**Review by fcatalan92 — APPROVED** [review]

**Pull request merged**